File: web/src/entities/entity.ts
export type position = { x: number; y: number };
export type size = { width: number; height: number };

export class Entity {
  prevPosition: position;

  constructor(
    public id: number,
    public position: position,
    public size: size,
    public color: string,
    public active: boolean,
  ) {}

  update(ctx: CanvasRenderingContext2D, offset: number, scale: number) {
    if (!this.active) return;

    ctx.fillStyle = this.color;
    ctx.fillRect(
      this.position.x * scale + offset,
      this.position.y * scale,
      this.size.width * scale,
      this.size.height * scale,
    );
  }
}

File: internal/entity/entity.go
package entity

type Entity struct {
	Id     int     `json:"id"`
	X      float64 `json:"x"`
	Y      float64 `json:"y"`
	Width  float64 `json:"w"`
	Height float64 `json:"h"`

	PrevX float64 `json:"-"`
	PrevY float64 `json:"-"`
}
